src/CMakeLists.txt

```diff
@@ -0,0 +1,218 @@
+      CudaFDKAlgorithm3D.cpp
+      CudaSirtAlgorithm3D.cpp
+      CudaBackProjectionAlgorithm3D.cpp
+      CudaForwardProjectionAlgorithm3D.cpp
+      Float32Data3DGPU.cpp
+      Float32ProjectionData3DGPU.cpp
+      Float32VolumeData3DGPU.cpp
+      "${CMAKE_SOURCE_DIR}/cuda/2d/algo.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/par_fp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/par_bp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/fan_fp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/fan_bp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/fbp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/sirt.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/sart.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/cgls.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/em.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/astra.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/util.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/arith.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/fft.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/darthelper.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/darthelper3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/algo3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/cgls3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/cone_fp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/cone_bp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/fdk.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/par3d_fp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/par3d_bp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/sirt3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/astra3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/util3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/arith3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/mem3d.cu")
+
+  set(ASTRA_CUDA_INCLUDES ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
+
+  # Set CUDA target architectures --------------------------------------------
+
+  if(NOT DEFINED ENV{CUDAARCHS})
+    set(CMAKE_CUDA_ARCHITECTURES "")
+    if((CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 5)
+       AND (CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 11))
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "30;32")
+    endif()
+    if((CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 5)
+       AND (CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 12))
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "35;37")
+    endif()
+    if((CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 6)
+       AND (CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 12))
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "50;52;53")
+    endif()
+    if(CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 8)
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "60;61;62")
+    endif()
+    if(CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 9)
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "70;72")
+    endif()
+    if(CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "75")
+    endif()
+    if(CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 11.1)
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "80;86")
+    endif()
+  endif()
+
+  message(
+    STATUS
+      "CUDA ${CMAKE_CUDA_COMPILER_VERSION} built for archs ${CMAKE_CUDA_ARCHITECTURES}"
+  )
```

Comment:
This decision tree is out of date and not really needed anymore! Not only does it not include the newest archs, but recent versions of CMake and NVCC allow you to pass "all-major", "all", or "native" as the target architecture where "all" is what this tree tries to emulate.